jina/serve/runtimes/worker/http_sagemaker_app.py

```diff
@@ -0,0 +1,178 @@
+
+if docarray_v2:
+    from docarray import BaseDoc, DocList
+
+
+def get_fastapi_app(
+    request_models_map: Dict,
+    caller: Callable,
+    logger: 'JinaLogger',
+    cors: bool = False,
+    **kwargs,
+):
+    """
+    Get the app from FastAPI as the REST interface.
+
+    :param request_models_map: Map describing the endpoints and its Pydantic models
+    :param caller: Callable to be handled by the endpoints of the returned FastAPI app
+    :param logger: Logger object
+    :param cors: If set, a CORS middleware is added to FastAPI frontend to allow cross-origin access.
+    :param kwargs: Extra kwargs to make it compatible with other methods
+    :return: fastapi app
+    """
+    with ImportExtensions(required=True):
+        from fastapi import FastAPI, Response, HTTPException
+        import pydantic
+        from fastapi.middleware.cors import CORSMiddleware
+
+    import os
+
+    from pydantic import BaseModel, Field
+    from pydantic.config import BaseConfig, inherit_config
+
+    from jina.proto import jina_pb2
+    from jina.serve.runtimes.gateway.models import _to_camel_case
+
+    class Header(BaseModel):
+        request_id: Optional[str] = Field(
+            description='Request ID', example=os.urandom(16).hex()
+        )
+
+        class Config(BaseConfig):
+            alias_generator = _to_camel_case
+            allow_population_by_field_name = True
+
+    class InnerConfig(BaseConfig):
+        alias_generator = _to_camel_case
+        allow_population_by_field_name = True
+
+    app = FastAPI()
+
+    if cors:
+        app.add_middleware(
+            CORSMiddleware,
+            allow_origins=['*'],
+            allow_credentials=True,
+            allow_methods=['*'],
+            allow_headers=['*'],
+        )
+        logger.warning('CORS is enabled. This service is accessible from any website!')
+
+    def add_post_route(
+        endpoint_path,
+        input_model,
+        output_model,
+        input_doc_list_model=None,
+        output_doc_list_model=None,
+    ):
+        app_kwargs = dict(
+            path=f'/{endpoint_path.strip("/")}',
+            methods=['POST'],
+            summary=f'Endpoint {endpoint_path}',
+            response_model=output_model,
+        )
+        if docarray_v2:
+            from docarray.base_doc.docarray_response import DocArrayResponse
+
+            app_kwargs['response_class'] = DocArrayResponse
+
+        @app.api_route(**app_kwargs)
+        async def post(body: input_model, response: Response):
+            req = DataRequest()
+            if body.header is not None:
+                req.header.request_id = body.header.request_id
+
+            if body.parameters is not None:
+                req.parameters = body.parameters
+            req.header.exec_endpoint = endpoint_path
+            data = body.data
+            if isinstance(data, list):
+                if not docarray_v2:
+                    req.data.docs = DocumentArray.from_pydantic_model(data)
```

Comment:
feel free to support only docarrayV2 if u want

Reply:
Done - https://github.com/jina-ai/jina/pull/6046/commits/5801568ea52c1882c6069574e7f2cf413be3bc04